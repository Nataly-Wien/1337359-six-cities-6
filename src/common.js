import {EMPTY_AVATAR_URL} from './const';

export const toUpperCaseFirst = (word) => word[0].toUpperCase() + word.slice(1);

export const ratingStyle = (rating) => {
  return {width: `${Math.round(rating) * 20}%`};
};

export const getPoints = (hotels = []) => hotels.map((item) => ({id: item.id, location: item.location, title: item.title}));

export const adaptUserToClient = (person) => {
  const clientPerson = {
    ...person,
    isPro: person.is_pro,
    avatarUrl: person.avatar_url ? person.avatar_url : EMPTY_AVATAR_URL,
  };

  delete clientPerson.is_pro;
  delete clientPerson.avatar_url;

  return clientPerson;
};

export const adaptCommentToClient = (comment) => {
  const clientComment = {
    ...comment,
    user: adaptUserToClient(comment.user),
  };

  delete clientComment.is_pro;
  delete clientComment.avatar_url;

  return clientComment;
};

export const adaptHotelToClient = (hotel) => {
  const clientHotel = {
    ...hotel,
    isFavorite: hotel.is_favorite,
    isPremium: hotel.is_premium,
    maxAdults: hotel.max_adults,
    previewImage: hotel.preview_image,
    host: adaptUserToClient(hotel.host),
    city: {
      ...hotel.city,
      location: {...hotel.city.location}
    },
    location: {...hotel.location},
    goods: hotel.goods.slice(),
    images: hotel.images.slice(),
  };

  delete clientHotel.is_favorite;
  delete clientHotel.is_premium;
  delete clientHotel.max_adults;
  delete clientHotel.preview_image;

  return clientHotel;
};
