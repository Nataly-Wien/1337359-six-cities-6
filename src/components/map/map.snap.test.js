import React from 'react';
import {render} from '@testing-library/react';
import Map from './map';
import {getPoints} from '../../common';

describe(`Should map component work correctly`, () => {
  it(`Should map at main page work correctly`, () => {

    const active = 1;
    const hotels = [
      {
        id: 5,
        city: {
          name: `Amsterdam`,
          location: {
            latitude: 52.37454,
            longitude: 4.897976,
            zoom: 12
          },
        },
        title: `Cityden Residences Museum Square`,
        location: {
          latitude: 52.3909553943508,
          longitude: 4.85309666406198,
          zoom: 16
        },
      },
      {
        id: 2,
        city: {
          name: `Amsterdam`,
          location: {
            latitude: 52.37454,
            longitude: 4.897976,
            zoom: 12
          },
        },
        title: `Dikker & Thijs Hotel`,
        location: {
          latitude: 52.369553943508,
          longitude: 4.85309666406198,
          zoom: 16
        },
      },
      {
        id: 1,
        city: {
          name: `Amsterdam`,
          location: {
            latitude: 52.37454,
            longitude: 4.897976,
            zoom: 12
          },
        },
        title: `Hotel van de Vijsel`,
        location: {
          latitude: 52.3345345,
          longitude: 4.834534,
          zoom: 16
        },
      },
      {
        id: 6,
        city: {
          name: `Amsterdam`,
          location: {
            latitude: 52.37454,
            longitude: 4.897976,
            zoom: 12
          },
        },
        title: `Park Plaza Victoria Amsterdam`,
        location: {
          latitude: 52.391908,
          longitude: 4.85978978,
          zoom: 16
        },
      },
    ];
    const {container} = render(<Map city={hotels.length > 0 ? hotels[0].city.location : {}}
      points={getPoints(hotels)} activeMarker={active} />);

    expect(container).toMatchSnapshot();
  });

  it(`Should map at room page work correctly`, () => {
    const currentHotel = {
      id: 1,
      city: {
        name: `Amsterdam`,
        location: {
          latitude: 52.37454,
          longitude: 4.897976,
          zoom: 12
        },
      },
      title: `Hotel van de Vijsel`,
      location: {
        latitude: 52.3345345,
        longitude: 4.834534,
        zoom: 16
      },
    };

    const {city, id, location, title} = currentHotel;

    const nearPlacesHotels = [
      {
        id: 5,
        title: `Cityden Residences Museum Square`,
        location: {
          latitude: 52.3909553943508,
          longitude: 4.85309666406198,
          zoom: 16
        },
      },
      {
        id: 2,
        title: `Dikker & Thijs Hotel`,
        location: {
          latitude: 52.369553943508,
          longitude: 4.85309666406198,
          zoom: 16
        },
      },
      {
        id: 6,
        title: `Park Plaza Victoria Amsterdam`,
        location: {
          latitude: 52.391908,
          longitude: 4.85978978,
          zoom: 16
        },
      },
    ];
    const {container} = render(<Map city={city.location}
      points={getPoints(nearPlacesHotels, {id, location, title})} activeMarker={id} />);

    expect(container).toMatchSnapshot();
  });
});
