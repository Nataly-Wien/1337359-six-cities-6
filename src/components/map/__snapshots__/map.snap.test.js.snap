// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should map component work correctly Should map at main page work correctly 1`] = `
<div>
  <div
    class="leaflet-container leaflet-grab leaflet-touch-drag"
    id="map"
    style="height: 100%; width: 100%; position: relative;"
    tabindex="0"
  >
    <div
      class="leaflet-pane leaflet-map-pane"
      style="left: 0px; top: 0px;"
    >
      <div
        class="leaflet-pane leaflet-tile-pane"
      >
        <div
          class="leaflet-layer "
          style="z-index: 1;"
        >
          <div
            class="leaflet-tile-container leaflet-zoom-animated"
            style="z-index: 18; left: 0px; top: 0px;"
          >
            <img
              alt=""
              class="leaflet-tile"
              role="presentation"
              src="https://c.basemaps.cartocdn.com/rastertiles/voyager/12/2103/1346.png"
              style="width: 256px; height: 256px; left: -186px; top: -5px;"
            />
          </div>
        </div>
      </div>
      <div
        class="leaflet-pane leaflet-shadow-pane"
      />
      <div
        class="leaflet-pane leaflet-overlay-pane"
      />
      <div
        class="leaflet-pane leaflet-marker-pane"
      >
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="img/pin.svg"
          style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: -130px; top: -78px; z-index: -78;"
          tabindex="0"
          title="Cityden Residences Museum Square"
        />
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="img/pin.svg"
          style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: -130px; top: 24px; z-index: 24;"
          tabindex="0"
          title="Dikker & Thijs Hotel"
        />
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="img/pin-active.svg"
          style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: -184px; top: 191px; z-index: 191;"
          tabindex="0"
          title="Hotel van de Vijsel"
        />
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="img/pin.svg"
          style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: -111px; top: -83px; z-index: -83;"
          tabindex="0"
          title="Park Plaza Victoria Amsterdam"
        />
      </div>
      <div
        class="leaflet-pane leaflet-tooltip-pane"
      />
      <div
        class="leaflet-pane leaflet-popup-pane"
      />
    </div>
    <div
      class="leaflet-control-container"
    >
      <div
        class="leaflet-top leaflet-left"
      />
      <div
        class="leaflet-top leaflet-right"
      />
      <div
        class="leaflet-bottom leaflet-left"
      />
      <div
        class="leaflet-bottom leaflet-right"
      >
        <div
          class="leaflet-control-attribution leaflet-control"
        >
          <a
            href="https://leafletjs.com"
            title="A JS library for interactive maps"
          >
            Leaflet
          </a>
           | © 
          <a
            href="https://www.openstreetmap.org/copyright"
          >
            OpenStreetMap
          </a>
           contributors © 
          <a
            href="https://carto.com/attributions"
          >
            CARTO
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Should map component work correctly Should map at room page work correctly 1`] = `
<div>
  <div
    class="leaflet-container leaflet-grab leaflet-touch-drag"
    id="map"
    style="height: 100%; width: 100%; position: relative;"
    tabindex="0"
  >
    <div
      class="leaflet-pane leaflet-map-pane"
      style="left: 0px; top: 0px;"
    >
      <div
        class="leaflet-pane leaflet-tile-pane"
      >
        <div
          class="leaflet-layer "
          style="z-index: 1;"
        >
          <div
            class="leaflet-tile-container leaflet-zoom-animated"
            style="z-index: 18; left: 0px; top: 0px;"
          >
            <img
              alt=""
              class="leaflet-tile"
              role="presentation"
              src="https://c.basemaps.cartocdn.com/rastertiles/voyager/12/2103/1346.png"
              style="width: 256px; height: 256px; left: -186px; top: -5px;"
            />
          </div>
        </div>
      </div>
      <div
        class="leaflet-pane leaflet-shadow-pane"
      />
      <div
        class="leaflet-pane leaflet-overlay-pane"
      />
      <div
        class="leaflet-pane leaflet-marker-pane"
      >
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="img/pin.svg"
          style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: -130px; top: -78px; z-index: -78;"
          tabindex="0"
          title="Cityden Residences Museum Square"
        />
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="img/pin.svg"
          style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: -130px; top: 24px; z-index: 24;"
          tabindex="0"
          title="Dikker & Thijs Hotel"
        />
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="img/pin.svg"
          style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: -111px; top: -83px; z-index: -83;"
          tabindex="0"
          title="Park Plaza Victoria Amsterdam"
        />
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="img/pin-active.svg"
          style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: -184px; top: 191px; z-index: 191;"
          tabindex="0"
          title="Hotel van de Vijsel"
        />
      </div>
      <div
        class="leaflet-pane leaflet-tooltip-pane"
      />
      <div
        class="leaflet-pane leaflet-popup-pane"
      />
    </div>
    <div
      class="leaflet-control-container"
    >
      <div
        class="leaflet-top leaflet-left"
      />
      <div
        class="leaflet-top leaflet-right"
      />
      <div
        class="leaflet-bottom leaflet-left"
      />
      <div
        class="leaflet-bottom leaflet-right"
      >
        <div
          class="leaflet-control-attribution leaflet-control"
        >
          <a
            href="https://leafletjs.com"
            title="A JS library for interactive maps"
          >
            Leaflet
          </a>
           | © 
          <a
            href="https://www.openstreetmap.org/copyright"
          >
            OpenStreetMap
          </a>
           contributors © 
          <a
            href="https://carto.com/attributions"
          >
            CARTO
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
